src/uct/ib/base/ib_iface.c

```diff
@@ -667,6 +667,60 @@ uct_ib_iface_roce_is_reachable(const uct_ib_device_gid_info_t *local_gid_info,
     return ret;
 }
 
+static int
+uct_ib_iface_is_same_device(const uct_ib_address_pack_params_t *params,
+                            uct_ib_iface_t *iface)
+{
+    const union ibv_gid *gid = &iface->gid_info.gid;
+    int res = 1;
+
+    if ((params->flags & UCT_IB_ADDRESS_PACK_FLAG_ETH) !=
+        uct_ib_iface_is_roce(iface)) {
+        return 0;
+    }
+
+    if (params->flags & UCT_IB_ADDRESS_PACK_FLAG_ETH) {
+        res &= !memcmp(gid->raw, params->gid.raw, sizeof(params->gid.raw));
```

Comment:
bitwise &= is wrong here

Reply:
can you explain why?

Reply:
it should be logical && with shorthand compute
something like 
if (memcmp(...)) return 0;